Add entity reference doc. (#1537)
o3de/o3de.org: content/docs/user-guide/components/reference/atom/entity-reference.md

toc: true
---

> The **Entity Reference** component allows tracking one or more entity IDs that can be conveniently accessed through script or C++ code during editor mode.

Reviewer: Switch to active voice - answers _who_ is doing the tracking and accessing of entity IDs. Users have to set this up and do this. 
```suggestion
The **Entity Reference** component allows you to track one or more entity IDs that you can conveniently access through script or C++ code during editor mode.
```

## Properties

![Entity Reference Component UI](/images/user-guide/components/reference/atom/entity-reference-ui.png)

| Property | Description |
| - | - |

> | **EntityIdReferences** | Contains a list of entities whose ids will be tracked |

Reviewer: ```suggestion
| **EntityIdReferences** | Contains a list of entities whose IDs will be tracked. |
```

## Usage ##

> During editor mode, call the EntityReferenceRequestBus to retrieve the list of entity being referenced by the component.

Reviewer: ```suggestion
During editor mode, call the `EntityReferenceRequestBus` to retrieve the list of entities being referenced by the component.
```

> This example uses the editor's python console to retrieve a list of entity ids.

Reviewer: ```suggestion
This example uses the O3DE's Python Console to retrieve a list of entity IDs.
```

> ```

Reviewer: Add python syntax highlighting by adding "python" after "```"
Will look like this: 
![image](https://user-images.githubusercontent.com/75444793/163273544-a2a80e50-0afa-4c4c-a193-45ce65fd0e34.png)

As opposed to this:
![image](https://user-images.githubusercontent.com/75444793/163273570-7e92fdc5-721c-4215-bdb2-b5cf38a331ef.png)

Author: Awesome suggestion, done!
